modules/nf-core/iqtree/main.nf

```diff
@@ -4,34 +4,88 @@ process IQTREE {
 
     conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/iqtree:2.3.0--h21ec9f0_0' :
-        'biocontainers/iqtree:2.3.0--h21ec9f0_0' }"
+        'https://depot.galaxyproject.org/singularity/iqtree:2.3.3--h21ec9f0_0' :
+        'biocontainers/iqtree:2.3.3--h21ec9f0_0' }"
 
     input:
     tuple val(meta), path(alignment)
-    val constant_sites
+    tuple val(meta), path(tree)
+    tuple val(meta), path(tree_te)
+    tuple val(meta), path(lmclust)
+    tuple val(meta), path(mdef)
+    tuple val(meta), path(partitions_equal)
+    tuple val(meta), path(partitions_proportional)
+    tuple val(meta), path(partitions_unlinked)
+    tuple val(meta), path(guide_tree)
+    tuple val(meta), path(sitefreq_in)
+    tuple val(meta), path(constraint_tree)
+    tuple val(meta), path(trees_z)
+    tuple val(meta), path(suptree)
+    tuple val(meta), path(trees_rf)
```

Comment:
I suggest removing all the meta's except the for the first one. Practically, files related to any given sample should be made into a tuple in some operation before it reaches this process, and will then be split into the channels with `map`/`multiMap`.

Reply:
I realized the way I had implemented the meta information did not make a lot of sense, will come up with a more appropriate solution.

Reply:
Fully removing the meta from most inputs is an issue, because the first input is optional and not required.
I looked at the options again and I believe that at least one of the first two inputs _has_ to be supplied for the program to run. I just kept the meta for these two inputs; by default the first meta is used, but if it is not supplied, the second will be used instead.